### ISSUE TITLE
Cronicle local sendmail not working

### ISSUE THREAD

**Issue by passbe** [jhuckaby/Cronicle#50]
## Summary

When configuring Cronicle to use the local sendmail configuration the error log shows it still trying to connect to 127.0.0.1:25

## Steps to reproduce the problem

Ensure configuration block exists:

`       "mail_options": {
                "sendmail": true,
                "newline": "unix",
                "path": "/usr/bin/sendmail"
        },
`

Remove both `smtp_hostname` and `smtp_port` from config.json. Run any job with a failure email notification.

### Operating system and version?

Arch Linux

### Node.js version?

9.3.0

### Cronicle software version?

0.7.5

### Are you using a multi-server setup, or just a single server?

Single

### Are you using the filesystem as back-end storage, or S3/Couchbase?

File system

### Can you reproduce the crash consistently?

Yes

## Log Excerpts

`[1513474254.788][2017-12-17 11:30:54][bots][Error][error][mail][Failed to send e-mail for job: jjba3hsu701: me@domain.com: Error: connect ECONNREFUSED 127.0.0.1:25][{"text":"To: me@domain.com\nFrom: test@example\nSubject: ⚠️ Cronicle Job Failed: Test\n\nDate/Time: 2017/12/17 11:30:54 (GMT+10)\nEvent Title: Test\nCategory: General\nServer Target: All Servers\nPlugin: Shell Script\n\nJob ID: jjba3hsu701\nHostname: example\nPID: 1161\nElapsed Time: 0 seconds\nPerformance Metrics: (No metrics provided)\nAvg. Memory Usage: (Unknown)\nAvg. CPU Usage: (Unknown)\nError Code: 1\n\nError Description:\nScript exited with code: 1\n\nJob Details:\nhttp://10.0.0.32:3012/#JobDetails?id=jjba3hsu701\n\nJob Debug Log (199 bytes):\nhttp://10.0.0.32:3012/api/app/get_job_log?id=jjba3hsu701\n\nEdit Event:\nhttp://10.0.0.32:3012/#Schedule?sub=edit_event&id=ejba26dra05\n\nEvent Notes:\n(None)\n\nRegards,\nThe Cronicle Team\n"}]`

Sorry @jhuckaby after the previous issue it still looks like it wont create the correct transport.

**Comment by jhuckaby** [comment]
Hey @passbe, 

Sorry about that.  Looks like pixl-mail was semver'ed to a **very** old nodemailer version, which was likely preventing sendmail mode from working properly.  I just released pixl-mail v1.0.8, which should now pull in nodemailer v4.4.1.

If you could, please try updating your pixl-mail module and restarting your Cronicle:

```sh
cd /opt/cronicle
sudo bin/control.sh stop
sudo npm install pixl-mail
sudo bin/control.sh start
```

Let me know, thanks!

**Comment by jhuckaby** [comment]
I just released Cronicle v0.7.6, which includes this fix, among a few other things:

https://github.com/jhuckaby/Cronicle/releases/tag/v0.7.6

Performing a standard Cronicle upgrade should do the trick.

## Upgrading Cronicle

To upgrade Cronicle, you can use the built-in `upgrade` command:

```
/opt/cronicle/bin/control.sh upgrade
```

This will upgrade the app and all dependencies to the latest stable release, if a new one is available.  It will not affect your data storage, users, or configuration settings.  All those will be preserved and imported to the new version.  For multi-server clusters, you'll need to repeat this command on each server.

**Comment by passbe** [comment]
@jhuckaby thanks again for your prompt reply. Upgrading now allows sendmail. Cheers.
